doc/data-protocol.md

```diff
@@ -0,0 +1,545 @@
+| 3           | Data byte 0             |
+| 4           | Data byte 1             |
+| [...]       | [...]                   |
+| N+3         | Data byte N             |
+
+The length of the data field is variable
+
+# CAN Bus
+
+> **Note:** the CAN bus between the Engine Computer and the Flight Controller can be extended to additional boards in future developments
+
+CAN bus: CAN2.0 (FlexCAN) @1Mbit/s with termination 120$\Omega$ resistors on the Flight Controller side and on the Engine Computer side
+
+Base frame format (11 identifier bits)
+
+# Data protocol
+
+The term `ID` is used below to denote `identifier field` of the CAN Bus and byte 2 (`Frame ID`) of a frame over the serial links
+
+The `identifier field` is an 11 bits field but only the 8 lower bits are used to make the data on the CAN Bus as similar as possible to the data on the serial buses. The remaining `ID` values `100` - `7FF` are available for data transfer between on-board systems over the CAN Bus only and cannot be sent to the ground.
+
+Each message on the Telecommand link and the Telemetry link has a unique ID
+
+| ID              | Frame type | From              | To                |
+|-----------------|------------|-------------------|-------------------|
+| `0x00` - `0x0F` | TC         | Ground            | Engine Computer   |
+| `0x10` - `0x7F` | TM         | Engine Computer   | Ground            |
+| `0x80` - `0x8F` | TC         | Ground            | Flight Controller |
+| `0x90` - `0xEF` | TM         | Flight Controller | Ground            |
+| `0xF0` - `0xFF` | N/A        | N/A               | N/A               |
+
+The `ID` values `0xF0` - `0xFF` are left unused and will be used if new devices are added to the on-board CAN Bus
+
+## Telecommand
+
+The Telecommand link is a low data rate radio link over LoRa at 433 MHz.
+
+### Communication sequence
+
+![commands-sequence-no-ec](diagrams/command-sequence-no-ec.png)
+>Command sequence to the Rocket. Command addressed to the Flight Controller<br>
+The source file can be found in [doc/diagrams/command-sequence.xml](diagrams/command-sequence.xml) and edited using [diagrams.net](https://www.diagrams.net/) \[[desktop](https://github.com/jgraph/drawio-desktop/releases)]
+
+![commands-sequence-no-ec](diagrams/command-sequence-ec.png)
+>Command sequence to the Rocket. Command addressed to the Engine Controller<br>
+The source file can be found in [doc/diagrams/command-sequence.xml](diagrams/command-sequence.xml) and edited using [diagrams.net](https://www.diagrams.net/) \[[desktop](https://github.com/jgraph/drawio-desktop/releases)]
+
+### Commands description
+
+#### Engine Controller
+
+Commands from the Ground Station to the Engine Computer
+
+<table>
+<thead>
+  <tr>
+    <th>ID</th>
+    <th>Description</th>
+    <th>Data size</th>
+    <th>Data</th>
+    <th>Comment</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>0x00</td>
+    <td>Time sync<br>Can be sent by FC</td>
+    <td>-</td>
+    <td>-</td>
+    <td></td>
```

Comment:
Should this return something akin to a UNIX Time stamp (i.e time since 1970-01-01)? Perhaps encoded as an 8 byte microsecond value?

Reply:
Yes. Or something closer to the way the GNSS receiver & RTC handle the time to reduce the processing overhead. Not looked at it yet